tests/SimplifyTest.elm

```diff
@@ -164,6 +165,156 @@ a = 1
+                """module A exposing (..)
+a = List.foldl (always identity) x
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "The call to List.foldl will result in the initial accumulator"
+                            , details = [ "You can replace this call by the initial accumulator." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = always x
+"""
+                        ]
+        , test "should fully qualify if import missing" <|
+            \() ->
+                """module A exposing (..)
+a = List.foldl f x << Set.toList
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "To fold a set, you don't need to convert to a List"
+                            , details = [ "Using Set.foldl directly is meant for this exact purpose and will also be faster." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = Set.foldl f x
+"""
+                        ]
+        , test "should qualify if not exposed" <|
+            \() ->
+                """module A exposing (..)
+import Set exposing (toList)
+a = List.foldl f x << toList
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "To fold a set, you don't need to convert to a List"
+                            , details = [ "Using Set.foldl directly is meant for this exact purpose and will also be faster." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Set exposing (toList)
+a = Set.foldl f x
+"""
+                        ]
+        , test "should not qualify if directly imported (exposed) explicitly" <|
+            \() ->
+                """module A exposing (..)
+import Set exposing (foldl)
+a = List.foldl f x << Set.toList
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "To fold a set, you don't need to convert to a List"
+                            , details = [ "Using Set.foldl directly is meant for this exact purpose and will also be faster." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Set exposing (foldl)
+a = foldl f x
```

Comment:
It's not covered already and would be nice to add!

Reply:
Let's do it in another PR yes.

For (1), I think that maybe we can peek ahead using `withFullAst` once again :sweat_smile: 
(2 and 3) Ugh, yes, there's that as well... I think I might make an issue in the Elm compiler to forbid shadowing imports because they're pretty annoying? :sweat_smile: